libs/core/uagents_core/utils/communication.py

```diff
@@ -0,0 +1,87 @@
+import json
+from typing import Optional, Any
+from uuid import uuid4, UUID
+
+import requests
+import urllib.parse
+
+from uagents_core.crypto import Identity
+from uagents_core.envelope import Envelope
+from uagents_core.config import DEFAULT_AGENTVERSE_URL, DEFAULT_ALMANAC_API_PATH
+from uagents_core.logger import get_logger
+
+
+
+logger = get_logger("uagents_core.utils.communication")
+
+
+def lookup_endpoint_for_agent(agent_address: str, *, agentverse_url: Optional[str] = None) -> str:
```

Comment:
This is a simplified version of the resolver, which does not handle multiple endpoints, for example. Not sure whether it makes sense to include here.

Reply:
first version is fine as it is I guess, but follow up versions needs to improve and add support for multiple endpoints.